src/api/post-createpledge.js

```diff
@@ -1,16 +1,14 @@
 async function postPledge(pledgeDetails) {
     const url = `${import.meta.env.VITE_API_URL}/pledges/`;
+    const authToken = localStorage.getItem("token");
     const response = await fetch(url, {
         method: "POST", 
         headers: {
         "Content-Type": "application/json",
+        "Authorization": `Token ${authToken}`,
         },
         body: JSON.stringify({
-        "amount": pledgeDetails.amount,
-        "comment": pledgeDetails.comment,
-        "is_anonymous": pledgeDetails.isAnonymous,
-        "project": pledgeDetails.project,
-        "id": pledgeDetails.id,
+       ...pledgeDetails
```

Comment:
here I am using the spread operator (...) instead of every individual key. Because the pledgeDetails parameter already has the data we want we can make it a little more readable and reduce the chance of missing required information.